### ISSUE TITLE
clang-tidy: 'readability-misleading-indentation' confused by "if (...) LABEL: {...}" depending on linebreaks

### ISSUE THREAD

**Issue by joelkp** [llvm/llvm-project#68823]
I've had some false positives from `clang-tidy` when using `readability-misleading-indentation`. It's confused by one specific C idiom where a label is added between an `if` and its `{`, also depending on the use of linebreaks. Below is a file that does it, tested with `clang-tidy` 16.0.6.

```
int main() {
        goto LABEL;
        // ...
        if (0)
        LABEL: {
                // contents here don't matter
        }
        return 0;
}
```

The check in `clang-tidy` seems to view the indentation on the `return 0;` line as being one level deeper than it actually is.

```
clang-tidy tidythis.c -checks='readability-misleading-indentation'           
...
1 warning generated.
/home/joel/swproj/misc/tidythis.c:8:2: warning: misleading indentation: statement is indented too deeply [readability-misleading-indentation]
        return 0;
        ^
/home/joel/swproj/misc/tidythis.c:3:2: note: did you mean this line to be inside this 'if'
        if (0)
        ^

```

Note that removing the linebreak after `if (0)` in this example, joining the two lines, will remove the false positive. As will the moving of `LABEL:` to just after `{`, instead of before it. And fully removing the use of a label also removes the warning.
